src/fsharp/FSharp.DependencyManager.Nuget/FSharp.DependencyManager.fs

```diff
@@ -176,31 +176,40 @@ type FSharpDependencyManager (outputDirectory:string option) =
 
     let key = "nuget"
     let name = "MsBuild Nuget DependencyManager"
-    let workingDirectory =
-        let path = Path.Combine(Path.GetTempPath(), key, Process.GetCurrentProcess().Id.ToString() + "--"+ Guid.NewGuid().ToString())
-        match outputDirectory with
-        | None -> path
-        | Some v -> Path.Combine(path, v)
 
     let generatedScripts = ConcurrentDictionary<string,string>()
 
+    let workingDirectory =
+        // Calculate the working directory for dependency management
+        //   if a path wasn't supplied to the dependency manager then use the temporary directory as the root
+        //   if a path was supplied if it was rooted then use the rooted path as the root
+        //   if the path wasn't supplied or not rooted use the temp directory as the root.
+        let directory =
+            let path = Path.Combine(Process.GetCurrentProcess().Id.ToString() + "--"+ Guid.NewGuid().ToString())
+            match outputDirectory with
+            | None -> Path.Combine(Path.GetTempPath(), path)
+            | Some v ->
+                if Path.IsPathRooted(v) then Path.Combine(v, path)
+                else Path.Combine(Path.GetTempPath(), path)
+
+        lazy
+            try
+                if not (Directory.Exists(directory)) then
+                    Directory.CreateDirectory(directory) |> ignore
+                directory
+            with | _ -> directory
+
     let deleteScripts () =
         try
-#if !Debug
-            if Directory.Exists(workingDirectory) then
-                Directory.Delete(workingDirectory, true)
+#if !DEBUG
+            if workingDirectory.IsValueCreated then
```

Comment:
Don't you only want to delete the directory if line 198 above (`Directory.CreateDirectory(directory) |> ignore`) was executed, not simply if the `lazy` block was entered?

Reply:
@brettfo , I think the real issue, is that the directory must be created.  If a directory with the name already exists, then it's still a problem.  Of course, given that the directory name is 'managed threadid - some randomly generated guid' then it's unlikely to already exist.  But I shall address that.